addons/hr_holidays/wizard/hr_departure_wizard.py

```diff
@@ -14,17 +13,35 @@ def action_register_departure(self):
         employee_leaves = self.env['hr.leave'].search([
             ('employee_id', '=', self.employee_id.id),
             ('date_to', '>', self.departure_date),
-            ('state', '!=', 'refuse')
         ])
-        to_delete = self.env['hr.leave']
-        to_cancel = self.env['hr.leave']
-        for leave in employee_leaves:
-            if leave.state == 'validate':
-                to_cancel |= leave
-            else:
-                to_delete |= leave
-        to_delete.unlink()
-        to_cancel._force_cancel(_('The employee no longer works in the company'), notify_responsibles=False)
+
+        employee_leaves_departure_date = employee_leaves.filtered(lambda leave: leave.date_from.date() <= self.departure_date)
```

Comment:
from what I get, you can also fetch the leaves with a date_to after the departure date. Once looping over them, either date_from is before that date and you split them, or you add them to the recordset to cancel, and cancel the second parts of the split as well :thinking: Note that we have a `_split_leave` method as well :smile: